Solution2.java

```diff
@@ -0,0 +1,77 @@
+// Time Complexity : O(MN)
+// Space Complexity : :O(MN)
```

Comment:
Good Job! but error in space complexity. Think if you are using any additional space here other than the data structure you get as input or give out as output.